### DISCUSSION TITLE
Drawer background scale setup wasnt mentioned in the docs

### DISCUSSION THREAD

**Discussion by nahasco** [shadcn-ui/ui#2905] [General]
To enable the nice background scale when the drawer opens, we have to wrap our app with 

```
<div vaul-drawer-wrapper="">
              <div className="relative flex min-h-screen flex-col bg-background">
                        {children}
              </div>
</div>
```

Although this feature was used in the [shadcn/ui ](https://ui.shadcn.com/) website and showcased in the example, it was never mentioned in the docs that we have to add these divs to make it work.

**Comment by 3li7u** [comment]
Well noted, Thanks @nahasco 
I noticed a strange behavior when the background gets scaled, some black background color is added to the `body` tag leading to a change in the theme color.
of course, I can add the `bg-background` class to the first child element in the `body` and it will be fixed, but I was wondering why this black background color was added

**Reply by marius-se** [reply]
Same here!

**Reply by DVGY** [reply]
same

**Reply by sronnaim** [reply]
How did you wrap it exactly?

Edit: Nevermind, Vaul does have setBackgroundColorOnScale on its Root component, so I went and edit there

**Reply by 4nur4g** [reply]
I faced the same issue of black background on opening the drawer. Is there a permanent fix without a workaround?

**Reply by DVGY** [reply]
> I faced the same issue of black background on opening the drawer. Is there a permanent fix without a workaround?

I don't think so. I have also used the same workround like this 
```
import { ReactNode } from "react";

export const DrawerCSSProvider = ({ children }: { children: ReactNode }) => {
  return (
    <div vaul-drawer-wrapper="">
      <div className="relative flex min-h-screen flex-col bg-background">
        {children}
      </div>
    </div>
  );
};

```
```
function App() {
  return (
    <ThemeProvider defaultTheme="dark" storageKey="vite-ui-theme">
      <DrawerCSSProvider>
        <Box className="w-[90vw] h-[90vh] overflow-scroll resize font-sans">
          <Appshell />
          <Toaster />
        </Box>
      </DrawerCSSProvider>
    </ThemeProvider>
  );
}

export default App;
```

**Reply by sronnaim** [reply]
> I faced the same issue of black background on opening the drawer. Is there a permanent fix without a workaround?

What I notice is that Drawer adds inline `background-color:black` styling to the `<body>` tag. Apparently `<Drawer>` component  generated from the CLI does have properties to costumize how the Drawer would fit your style, one of which is the property
` setBackgroundColorOnScale`. So I'd say to check there first.

**Reply by DVGY** [reply]
> > I faced the same issue of black background on opening the drawer. Is there a permanent fix without a workaround?
> 
> What I notice is that Drawer adds inline `background-color:black` styling to the `<body>` tag. Apparently `<Drawer>` component generated from the CLI does have properties to costumize how the Drawer would fit your style, one of which is the property ` setBackgroundColorOnScale`. So I'd say to check there first.

this also works `setBackgroundColorOnScale={false}`!!

**Comment by fmendez89** [comment]
Just to let you know, on newer versions the prop added to the div has changed.

vaul-drawer-wrapper now is data-vaul-drawer-wrapper

Refer to:
https://github.com/emilkowalski/vaul/issues/411
